File: Leetcode Problems/2109.py
class Solution:
    def addSpaces(self, s: str, spaces: list[int]) -> str:
        s = list(s)
        
        for i, pos in enumerate(spaces):
            s.insert(pos + i, ' ')
        
        return ''.join(s)

if __name__ == '__main__':
    s = "hello"
    spaces = [2, 3]
    print(Solution().addSpaces(s, spaces))

File: Leetcode Problems/2109.cpp
#include<iostream>
#include<vector>
#include<string>
using namespace std;

class Solution {
public:
    string addSpaces(string s, vector<int>& spaces) {
        vector<char> ans;
        int a = 0;

        for(int i=0; i<spaces.size(); i++) {
            int b = spaces[i];

            for(int i=a; i<b; i++) {
                ans.push_back(s[i]);
            }
            ans.push_back(' ');

            a = b;
        }

        for(int i=a; i<s.size(); i++) {
            ans.push_back(s[i]);
        }

        string res(ans.begin(), ans.end());

        return res;
    }
};

int main() {
    string s = "ABC";
    vector<int> spaces = {1, 2};
    Solution obj;
    cout << obj.addSpaces(s, spaces) << endl;
    return 0;
}